projects/packages/identity-crisis/src/class-rest-endpoints.php

```diff
@@ -0,0 +1,164 @@
+			'jetpack/v4',
+			'/identity-crisis/confirm-safe-mode',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::confirm_safe_mode',
+				'permission_callback' => __CLASS__ . '::identity_crisis_mitigation_permission_check',
+			)
+		);
+
+		// Handles the request to migrate stats and subscribers during an identity crisis.
+		register_rest_route(
+			'jetpack/v4',
+			'identity-crisis/migrate',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::migrate_stats_and_subscribers',
+				'permission_callback' => __CLASS__ . '::identity_crisis_mitigation_permission_check',
+			)
+		);
+
+		// IDC resolve: create an entirely new shadow site for this URL.
+		register_rest_route(
+			'jetpack/v4',
+			'/identity-crisis/start-fresh',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::start_fresh_connection',
+				'permission_callback' => __CLASS__ . '::identity_crisis_mitigation_permission_check',
+			)
+		);
+
+	}
+
+	/**
+	 * Handles identity crisis mitigation, confirming safe mode for this site.
+	 *
+	 * @since 4.4.0
+	 *
+	 * @return bool | WP_Error True if option is properly set.
+	 */
+	public static function confirm_safe_mode() {
+		$updated = Jetpack_Options::update_option( 'safe_mode_confirmed', true );
+		if ( $updated ) {
+			return rest_ensure_response(
+				array(
+					'code' => 'success',
+				)
+			);
+		}
+
+		return new WP_Error(
+			'error_setting_jetpack_safe_mode',
+			esc_html__( 'Could not confirm safe mode.', 'jetpack' ),
+			array( 'status' => 500 )
+		);
+	}
+
+	/**
+	 * Handles identity crisis mitigation, migrating stats and subscribers from old url to this, new url.
+	 *
+	 * @since 4.4.0
+	 *
+	 * @return bool | WP_Error True if option is properly set.
+	 */
+	public static function migrate_stats_and_subscribers() {
+		if ( Jetpack_Options::get_option( 'sync_error_idc' ) && ! Jetpack_Options::delete_option( 'sync_error_idc' ) ) {
+			return new WP_Error(
+				'error_deleting_sync_error_idc',
+				esc_html__( 'Could not delete sync error option.', 'jetpack' ),
+				array( 'status' => 500 )
+			);
+		}
+
+		if ( Jetpack_Options::get_option( 'migrate_for_idc' ) || Jetpack_Options::update_option( 'migrate_for_idc', true ) ) {
+			return rest_ensure_response(
+				array(
+					'code' => 'success',
+				)
+			);
+		}
+		return new WP_Error(
+			'error_setting_jetpack_migrate',
+			esc_html__( 'Could not confirm migration.', 'jetpack' ),
+			array( 'status' => 500 )
+		);
+	}
+
+	/**
+	 * This IDC resolution will disconnect the site and re-connect to a completely new
+	 * and separate shadow site than the original.
+	 *
+	 * It will first will disconnect the site without phoning home as to not disturb the production site.
+	 * It then builds a fresh connection URL and sends it back along with the response.
+	 *
+	 * @since 4.4.0
+	 * @return bool|WP_Error
```

Comment:
I think this method returns a `WP_REST_Response` or a `WP_Error`.

Reply:
updated.